acasclient/CmpdReg.py

```diff
@@ -0,0 +1,196 @@
+
+    # Check for salts
+    salt_abrevs = [iso_salt['salt']["abbrev"] for iso_salt in  meta_lot["isosalts"]]
+    salt_equivs = [str(iso_salt["equivalents"]) for iso_salt in  meta_lot["isosalts"]]
+
+    # Synthesis date is in the format MM/DD/YYYY
+    # However Creg isn't good at converting the date back to the correct format
+    # So we need to convert it to YYYY-MM-DD
+    synthesis_date = meta_lot["lot"]["synthesisDate"]
+    if synthesis_date is not None:
+        synthesis_date = synthesis_date[6:] + '-' + synthesis_date[:2] + '-' + synthesis_date[3:5]
+
+    # If lot amount is filled in, then we need to fill in lot barcode with the lot corp name
+    lot_barcode = meta_lot["lot"]["corpName"] if meta_lot["lot"]["amount"] is not None else None
+
+    return_dict = {
+        'mol': meta_lot["lot"]['asDrawnStruct'] if meta_lot["lot"]['asDrawnStruct'] is not None else meta_lot["lot"]["parent"]["molStructure"],
+        "id": meta_lot["lot"]["id"],
+        "name": meta_lot["lot"]["corpName"],
+        "parent_common_name": '; '.join(parent_common_names) if len(parent_common_names) > 0 else None,
+        "parent_corp_name": meta_lot["lot"]["parent"]["corpName"],
+        "lot_amount":  meta_lot["lot"]["amount"],
+        "lot_amount_units":  meta_lot["lot"]["amountUnits"]["code"] if meta_lot["lot"]["amountUnits"] is not None else None,
+        "lot_color": meta_lot["lot"]["color"],
+        "lot_synthesis_date": synthesis_date,
+        "lot_notebook_page": meta_lot["lot"]["notebookPage"],
+        "lot_corp_name": meta_lot["lot"]["corpName"],
+        "lot_number": meta_lot["lot"]["lotNumber"],
+        "lot_purity": meta_lot["lot"]["purity"],
+        "lot_purity_opertator": meta_lot["lot"]["purityOperator"]["code"] if meta_lot["lot"]["purityOperator"] is not None else None,
+        "lot_comments": meta_lot["lot"]["comments"],
+        "lot_chemist": meta_lot["lot"]["chemist"],
+        "lot_solution_amount":  meta_lot["lot"]["solutionAmount"],
+        "lot_solution_amount_units": meta_lot["lot"]["solutionAmountUnits"]["code"] if meta_lot["lot"]["solutionAmountUnits"] is not None else None,
+        "lot_supplier": meta_lot["lot"]["supplier"],
+        "lot_supplier_id": meta_lot["lot"]["supplierID"],
+        "project": meta_lot["lot"]["project"],
+        "parent_stereo_category": meta_lot["lot"]["parent"]["stereoCategory"]['code'],
+        "parent_stereo_comment": meta_lot["lot"]["parent"]["stereoComment"],
+        "lot_is_virtual": meta_lot["lot"]["isVirtual"],
+        "lot_supplier_lot": meta_lot["lot"]["supplierLot"],
+        "lot_salt_abbrev": '; '.join(salt_abrevs) if len(salt_abrevs) > 0 else None,
+        "lot_salt_equivalents": '; '.join(salt_equivs) if len(salt_equivs) > 0 else None,
+        "lot_barcode": lot_barcode
+    }
+    return return_dict
+
+
+def convert_dict_to_type(dict, type_name):
+    """Converts a dictionary to a type."""
```

Comment:
Could use a little more context here on what this is used for, and what is meant by "type" here. Python type hints for the input arguments would be helpful.